crates/iota-sdk-ffi/src/types/graphql.rs

```diff
@@ -0,0 +1,319 @@
+    #[uniffi(default = None)]
+    pub at_checkpoint: Option<u64>,
+    #[uniffi(default = None)]
+    pub before_checkpoint: Option<u64>,
+    #[uniffi(default = None)]
+    pub sign_address: Option<Arc<Address>>,
+    #[uniffi(default = None)]
+    pub recv_address: Option<Arc<Address>>,
+    #[uniffi(default = None)]
+    pub input_object: Option<Arc<ObjectId>>,
+    #[uniffi(default = None)]
+    pub changed_object: Option<Arc<ObjectId>>,
+    #[uniffi(default = None)]
+    pub transaction_ids: Option<Vec<String>>,
+    #[uniffi(default = None)]
+    pub wrapped_or_deleted_object: Option<Arc<ObjectId>>,
+}
+
+impl From<iota_graphql_client::query_types::TransactionsFilter> for TransactionsFilter {
+    fn from(value: iota_graphql_client::query_types::TransactionsFilter) -> Self {
+        Self {
+            function: value.function,
+            kind: value.kind.map(Into::into),
+            after_checkpoint: value.after_checkpoint,
+            at_checkpoint: value.at_checkpoint,
+            before_checkpoint: value.before_checkpoint,
+            sign_address: value.sign_address.map(Into::into).map(Arc::new),
+            recv_address: value.recv_address.map(Into::into).map(Arc::new),
+            input_object: value.input_object.map(Into::into).map(Arc::new),
+            changed_object: value.changed_object.map(Into::into).map(Arc::new),
+            transaction_ids: value.transaction_ids,
+            wrapped_or_deleted_object: value
+                .wrapped_or_deleted_object
+                .map(Into::into)
+                .map(Arc::new),
+        }
+    }
+}
+
+impl From<TransactionsFilter> for iota_graphql_client::query_types::TransactionsFilter {
+    fn from(value: TransactionsFilter) -> Self {
+        Self {
+            function: value.function,
+            kind: value.kind.map(Into::into),
+            after_checkpoint: value.after_checkpoint,
+            at_checkpoint: value.at_checkpoint,
+            before_checkpoint: value.before_checkpoint,
+            sign_address: value.sign_address.map(|v| **v),
+            recv_address: value.recv_address.map(|v| **v),
+            input_object: value.input_object.map(|v| **v),
+            changed_object: value.changed_object.map(|v| **v),
+            transaction_ids: value.transaction_ids,
+            wrapped_or_deleted_object: value.wrapped_or_deleted_object.map(|v| **v),
+        }
+    }
+}
+
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct DryRunResult(pub iota_graphql_client::DryRunResult);
+
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct Event(pub iota_types::Event);
+
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct ObjectRef(pub iota_graphql_client::query_types::ObjectRef);
+
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct Epoch(pub iota_graphql_client::query_types::Epoch);
+
+#[derive(Clone, Debug, derive_more::From, uniffi::Record)]
```

Comment:
What does this `derive_more::From` do?

Reply:
accidentally left it after converting the wrapper impl

Reply:
whoops, done